lib/convert.js

```diff
@@ -1,60 +1,142 @@
+        var generateNormals = defaultValue(options.generateNormals, false);
+        var ao = defaultValue(options.ao, false);
+        var kmc = defaultValue(options.kmc, false);
+        var textureCompressionOptions = options.textureCompressionOptions;
+        var bypassPipeline = defaultValue(options.bypassPipeline, false);
+        var logger = defaultValue(options.logger, defaultLogger);
+        options.logger = logger;
+        options.hasTransparency = defaultValue(options.hasTransparency, false);
+        options.secure = defaultValue(options.secure, false);
+
+        if (!defined(objPath)) {
+            throw new DeveloperError('objPath is required');
+        }
+
+        if (!defined(gltfPath)) {
+            throw new DeveloperError('gltfPath is required');
+        }
+
+        var objExtension = path.extname(objPath).toLowerCase();
+        if (objExtension !== '.obj') {
+            throw new DeveloperError('Invalid obj path "' + objPath + '"');
+        }
+
+        var extension = path.extname(gltfPath).toLowerCase();
+        if (extension !== '.gltf' && extension !== '.glb') {
+            throw new DeveloperError('Invalid gltf path "' + gltfPath + '"');
+        }
+
+        var basePath = path.dirname(gltfPath);
+        var modelName = path.basename(gltfPath, path.extname(gltfPath));
+        if (extension === '.glb') {
+            binary = true;
+            if (bypassPipeline) {
+                logger('--bypassPipeline does not convert to binary glTF, saving as .gltf');
+                extension = '.gltf';
+            }
+        }
+        gltfPath = path.join(path.dirname(gltfPath), modelName + extension);
+
+        var aoOptions = ao ? {} : undefined;
+        var kmcOptions = kmc ? {} : undefined;
+
+        var pipelineOptions = {
+            createDirectory : false,
+            basePath : basePath,
+            binary : binary,
+            embed : !separate,
+            embedImage : !separateTextures,
+            quantize : compress,
+            compressTextureCoordinates : compress,
+            encodeNormals : compress,
+            preserve : !optimize,
+            optimizeForCesium : optimizeForCesium,
+            smoothNormals : generateNormals,
+            aoOptions : aoOptions,
+            kmcOptions : kmcOptions,
+            textureCompressionOptions : textureCompressionOptions
+        };
+
+        return loadObj(objPath, options)
+            .then(function(objData) {
+                return createGltf(objData);
+            })
+            .then(function(gltf) {
+                return writeUris(gltf, gltfPath, separate, separateTextures, logger);
+            })
+            .then(function(gltf) {
+                if (bypassPipeline) {
+                    return convert._outputJson(gltfPath, gltf);
+                } else {
+                    return GltfPipeline.processJSONToDisk(gltf, gltfPath, pipelineOptions);
+                }
+            })
+            .then(resolve)
```

Comment:
The entire outer promise is unneeded, since the function is synchronous up until the initial `loadObj` call.